astropy/units/format/fits.py

```diff
@@ -65,31 +65,33 @@ def _units(cls) -> dict[str, UnitBase]:
 
     @classmethod
     def to_string(
-        cls, unit: UnitBase, fraction: bool | Literal["inline", "multiline"] = False
+        cls,
+        unit: CompositeUnit | NamedUnit,
```

Comment:
This parameter makes the signature inconsistent with `Base.to_string()`.

Reply:
Much more importantly, the updated signature is inconsistent with `UnitBase.to_string()`. So this change is not solving a type error, it's just pushing the error to `astropy.units.core`, but you can't tell that because you are configuring the type checker to ignore that module for now.

Reply:
Hum, this seems like a strong argument to re-expand the initial target to the whole units subpackage. What do you think, @mhvk ?

Reply:
Indeed, perhaps we should go back to `astropy.units` as a whole. Though that beats my goal here of making it small enough that we can actually get some CI going in a short timeframe. (I quickly checked whether something else might work, but as of now, not even `astropy/units/utils.py` is OK...)

Reply:
How about starting with `stats` then (#16562) ?

Reply:
Running `mypy --follow-imports silent astropy/stats/` reports 471 errors. The exact number might be different if I'd install `scipy-stubs`, but even then I wouldn't get my hopes up. My understanding is that in `stats` type information from docstrings was rewritten as type annotations, but there were no attempts to ensure that the resulting annotations were correct, or even self-consistent, or even valid.

Running `mypy --follow-imports silent astropy/config/*.py` doesn't report any errors, but not everything in `config` is annotated and Mypy skips unannotated functions, so `basedpyright` might report many errors even in `config`.